### PR TITLE
pkg/ctime: Factor libpod/finished* into a separate package

### PR DESCRIPTION
**Pull request by wking**
This removes some boilerplate from the libpod package, so we can focus on container stuff there.  And it gives us a tidy sub-package for focusing on ctime extraction, so we can focus on unit testing and portability of the extraction utility there.

The unit test seems a bit flaky at the moment, with creation times that predate `begin`, at least on my dev box.

### CODE REVIEW

**Comment by mheon** [comment]
LGTM

**Comment by wking** [comment]
I've pushed caf5374ec -> 46feb2467, fixing [the gofmt issue][1].

[1]: https://s3.amazonaws.com/aos-ci/ghprb/projectatomic/libpod/697c22b1ed3d07e26aa92200923ebbfe7ae182ab.1.1530813692939354176/output.log

**Comment by wking** [comment]
I'm seeing the same flake [on Travis][1] that I'd seen locally:

```
ctime_test.go:34: created 2018-07-05 19:44:20.858799425 +0000 UTC is 3023811 nanoseconds before 2018-07-05 19:44:20.861823236 +0000 UTC m=+0.000492159
```

I'll play with this more tonight, but am happy for any suggestions folks have about what's going on there...

[1]: https://travis-ci.org/projectatomic/libpod/jobs/400559295#L1915

**Comment by mheon** [comment]
A bit of searching found https://stackoverflow.com/questions/14392975/timestamp-accuracy-on-ext4-sub-millsecond as a potential explanation

**Comment by mheon** [comment]
All we really want to test is that the file change time is (relatively) close to the time we get from the OS, indicating we're getting sane results. Taking the absolute value of the difference between the two and ensuring it's less than a microsecond or two sounds like a good way of doing that?

**Comment by wking** [comment]
> ... ensuring it's less than a microsecond or two...

In the failure quoted above, it was just over 3ms, which is over 1k times your limit ;).  And if the limit is the configurable kernel timer, it's not clear to me how far I should relax the comparison.

**Comment by wking** [comment]
More interesting background reading [in this thread][1].  The lack of interpolation is the point I was missing [above][2]: it's time-at-the-last-tick, not value-of-the-last-expensive-time-call.  Golang doesn't give us a way to select `CLOCK_REALTIME_COARSE`, so I've pushed 0ce4123e -> 9eba6632 to give a second of padding to `begin` and (to ensure monotonicity within the larger window) I'm testing two sequentially-created files.

[1]: https://groups.google.com/d/msg/linux.kernel/mdeXx2TBYZA/OyP7orFmAQAJ
[2]: https://github.com/projectatomic/libpod/pull/1050#issuecomment-402897890

**Comment by wking** [comment]
I've pushed 9eba6632 -> 2e93d0ff fixing a broken `package` declaration and changing the unsupported implementation to use [`ModTime()`][1] (more discussion in the commit message).

[1]: https://golang.org/pkg/os/#FileInfo

File: pkg/ctime/ctime_linux_32.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build arm 386
+// +build arm,linux 386,linux
 
 package libpod
```
Comment (wking) [inline pkg/ctime/ctime_linux_32.go]: Oops, still need to update this.

Comment (wking) [reply pkg/ctime/ctime_linux_32.go]: > Oops, still need to update this.

Fixed with 1b9f13f7 -> c3224bf2.

**Comment by wking** [comment]
[The Travis failure][1] is:

```
package golang.org/x/crypto/ssh/terminal: unrecognized import path "golang.org/x/crypto/ssh/terminal" (https fetch: Get https://golang.org/x/crypto/ssh/terminal?go-get=1: dial tcp 74.125.196.141:443: i/o timeout)
make: *** [.install.gitvalidation]
```

Just a flakey day today ;).

[1]: https://travis-ci.org/projectatomic/libpod/jobs/400903085#L96

**Comment by mheon** [comment]
bot, retest this please

**Comment by wking** [comment]
Everybody is happy except [the Travis flake][1].  I have another commit locally to drop the 32-/64-bit distinction on Linux (which seems to work locally).  Should I push that here to trigger retests?  Or wait until this lands and file it as a new PR?

[1]: https://github.com/projectatomic/libpod/pull/1050#issuecomment-403063802

**Comment by mheon** [comment]
@wking New PR once this lands is probably better.
Let me restart the failing travis jobs.

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**